docs/utilities/batch.md

```diff
@@ -1,460 +1,287 @@
-                         <goals>
-                             <goal>compile</goal>
-                         </goals>
-                     </execution>
-                 </executions>
-            </plugin>
-            ...
-        </plugins>
-    </build>
-    ```
-
-=== "Gradle Java 11+"
-
-    ```groovy hl_lines="3 11"
-        plugins {
-            id 'java'
-            id 'io.freefair.aspectj.post-compile-weaving' version '8.1.0'
-        }
-        
-        repositories {
-            mavenCentral()
-        }
-        
-        dependencies {
-            aspect 'software.amazon.lambda:powertools-sqs:{{ powertools.version }}'
-        }
-        
-        sourceCompatibility = 11 // or higher
-        targetCompatibility = 11 // or higher
     ```
 
-=== "Gradle Java 1.8"
+=== "Gradle"
 
-    ```groovy hl_lines="3 11"
-        plugins {
-            id 'java'
-            id 'io.freefair.aspectj.post-compile-weaving' version '6.6.3'
-        }
+    ```groovy
         
         repositories {
             mavenCentral()
         }
         
         dependencies {
-            aspect 'software.amazon.lambda:powertools-sqs:{{ powertools.version }}'
+            aspect 'software.amazon.lambda:powertools-batch:{{ powertools.version }}'
         }
-        
-        sourceCompatibility = 1.8
-        targetCompatibility = 1.8
     ```
 
 ## IAM Permissions
 
-This utility requires additional permissions to work as expected. Lambda functions using this utility require the `sqs:DeleteMessageBatch` permission.
-
-If you are also using [nonRetryableExceptions](#move-non-retryable-messages-to-a-dead-letter-queue) attribute, utility will need additional permission of `sqs:GetQueueAttributes` on source SQS. 
-It also needs `sqs:SendMessage` and `sqs:SendMessageBatch` on configured dead letter queue.  
+The [Lambda execution role](https://docs.aws.amazon.com/lambda/latest/dg/lambda-intro-execution-role.html) of your function 
+requires appropriate permissions for the message source you are using. In each case you should create a policy that restricts
+access to the queue, stream, or table, that your lambda is reading batches with.
 
-If source or dead letter queue is configured to use encryption at rest using [AWS Key Management Service (KMS)](https://aws.amazon.com/kms/), function will need additional permissions of 
-`kms:GenerateDataKey` and `kms:Decrypt` on the KMS key being used for encryption. Refer [docs](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-key-management.html#compatibility-with-aws-services) for more details.
-
-Refer [example project](https://github.com/aws-samples/aws-lambda-powertools-examples/blob/main/java/SqsBatchProcessing/template.yaml#L105) for policy details example.
+* **SQS** - `SQS:ReceiveMessage`, ``SQS::DeleteMessage``, ``SQS::GetQueueAttributes`` - [further details](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#events-sqs-permissions)
```

Comment:
@mriccia @jeromevdl [the documentation](https://docs.aws.amazon.com/lambda/latest/dg/with-sqs.html#events-sqs-permissions) claims as I understand it you need these perms. But - when I deploy our examples with SAM, it all works without them. This fits my mental model because the function itself isn't dealing with SQS, Lambda is.

What gives?

Reply:
did you look at the permissions of your function? does sam create them for you when you add a trigger ?

Reply:
Yes I did, and no it doesn't. I'll look again for all the variants as I am still very suspicious.